pkg/platform/runtime/buildscript/json.go

```diff
@@ -90,48 +83,74 @@ func (f *FuncCall) MarshalJSON() ([]byte, error) {
-				versionReqs, err := model.VersionStringToRequirements(value)
-				if err != nil {
-					return nil, errs.Wrap(err, "Could not parse version requirements")
-				}
+	requirement := make(map[string]interface{})
 
-				var scriptReqs []*Value
-				for _, req := range versionReqs {
-					var assignments []*Assignment
-					comparator, ok := req[bpModel.VersionRequirementComparatorKey]
-					if ok {
-						assignments = append(assignments, &Assignment{buildexpression.RequirementComparatorKey, &Value{Str: ptr.To(comparator)}})
-					}
+	for _, arg := range args {
+		assignment := arg.Assignment
+		if assignment == nil {
+			return nil, errors.New(fmt.Sprintf("Cannot marshal %v", arg))
+		}
 
-					version, ok := req[bpModel.VersionRequirementVersionKey]
-					if ok {
-						assignments = append(assignments, &Assignment{buildexpression.RequirementVersionKey, &Value{Str: ptr.To(version)}})
+		switch {
+		// Marshal the name argument (e.g. name = "<namespace>/<name>") into
+		// {"name": "<name>", "namespace": "<namespace>"}
+		case assignment.Key == buildexpression.RequirementNameKey && assignment.Value.Str != nil:
+			name, namespace := separateNamespace(*assignment.Value.Str)
+			requirement[buildexpression.RequirementNameKey] = name
+			requirement[buildexpression.RequirementNamespaceKey] = namespace
+
+		// Marshal the version argument (e.g. version = <op>("<version>")) into
+		// {"version_requirements": [{"comparator": "<op>", "version": "<version>"}]}
+		case assignment.Key == buildexpression.RequirementVersionKey && assignment.Value.FuncCall != nil:
+			var requirements []*Value
+			var addRequirement func(*FuncCall) error // recursive function for adding to requirements list
+			addRequirement = func(funcCall *FuncCall) error {
+				switch name := funcCall.Name; name {
+				case eqFuncName:
+					fallthrough
+				case neFuncName:
+					fallthrough
+				case gtFuncName:
+					fallthrough
+				case gteFuncName:
+					fallthrough
+				case ltFuncName:
+					fallthrough
+				case lteFuncName:
+					req := make([]*Assignment, 0)
+					req = append(req, &Assignment{buildexpression.RequirementComparatorKey, &Value{Str: ptr.To(strings.ToLower(name))}})
+					if len(funcCall.Arguments) == 0 || funcCall.Arguments[0].Str == nil {
+						return errors.New(fmt.Sprintf("Illegal argument for version comparator '%s': string expected", name))
 					}
-
-					reqValue := &Value{Object: &assignments}
-					scriptReqs = append(scriptReqs, reqValue)
+					req = append(req, &Assignment{buildexpression.RequirementVersionKey, &Value{Str: funcCall.Arguments[0].Str}})
+					requirements = append(requirements, &Value{Object: &req})
+				case andFuncName:
+					for _, a := range funcCall.Arguments {
+						if a.FuncCall == nil {
+							return errors.New(fmt.Sprintf("Illegal argument for version comparator '%s': function expected", name))
+						}
+						err := addRequirement(a.FuncCall)
+						if err != nil {
+							return err
```

Comment:
There are a couple errors in this function that need to be wrapped. May be worth skimming the changeset again to ensure we're wrapping everything.